runtime/doc/lsp.txt

```diff
@@ -173,6 +173,27 @@ name: >
   vim.lsp.protocol.TextDocumentSyncKind.Full == 1
   vim.lsp.protocol.TextDocumentSyncKind[1] == "Full"
 
+================================================================================
+LSP HIGHLIGHT GROUPS                                      *lsp-highlight-groups*
+
+                                                        *hl-LspDiagnosticsError*
+LspDiagnosticsError       used for "Error" diagnostic virtual text
+                                                      *hl-LspDiagnosticsWarning*
+LspDiagnosticsWarning     used for "Warning" diagnostic virtual text
+                                                  *hl-LspDiagnosticsInformation*
+LspDiagnosticInformation  used for "Information" diagnostic virtual text
+                                                         *hl-LspDiagnosticsHint*
+LspDiagnosticHint         used for "Hint" diagnostic virtual text
+                                                    *hl-LspDiagnosticsUnderline*
+LspDiagnosticsUnderline   used for text with diagnostic information
```

Comment:
The description could be enhanced: does this mean it applies to all previous highlights ? Why force an underline when it can be just a default ?

Reply:
`LspDiagnosticsUnderline` was not added by me, I just added the documentation since it was missing and I was adding the docs for the new highlights anyway.

`LspDiagnostics(Error|Warning|Hint|Information)` is applied to diagnostics (virtual text and sign column), but is not applied to "diagnosed" code/text. 
On the contrary, `LspDiagnosticsUnderline` is applied to the code/text. 

Maybe we should change `LspDiagnosticsUnderline` to `LspDiagnosedText` since people may want to customize it (eg: use italics instead of underline) /cc @norcalli

Reply:
LspDiagnosedText or LspDiagnosticAny/LspDiagnosticAll to keep the same prefix. You can either remove the doc for this specific highlight and do it in another PR or update all at once here.